fog/overseer/server/src/worker.rs

```diff
@@ -0,0 +1,543 @@
+                            log::info!(self.logger, "Automatic failover completed successfully.")
+                        }
+                        Err(err) => {
+                            log::error!(self.logger, "Automatic failover failed: {:?}", err)
+                        }
+                    };
+                }
+                1 => {
+                    log::trace!(
+                        self.logger,
+                        "There is one active node in the Fog Ingest cluster. Active ingress key: {:?}",
+                        active_ingest_summary_node_mappings[0]
+                        .ingest_summary
+                        .get_ingress_pubkey()
+                    );
+                    continue;
+                }
+                _ => {
+                    let active_node_ingress_pubkeys: Vec<&external::CompressedRistretto> =
+                        active_ingest_summary_node_mappings
+                            .iter()
+                            .map(|active_ingest_summary_node_mapping| {
+                                active_ingest_summary_node_mapping
+                                    .ingest_summary
+                                    .get_ingress_pubkey()
+                            })
+                            .collect();
+                    log::warn!(
+                        self.logger,
+                        "There are multiple active nodes in the Fog Ingest cluster. Active ingress keys: {:?}",
+                        active_node_ingress_pubkeys
+                    );
+                    // TODO: Set up sentry alerts and signal to ops that two
+                    // keys are active at once. This is
+                    // unexpected.
+                }
+            }
+        }
+    }
+
+    /// Returns the latest round of ingest summaries for each
+    /// FogIngestGrpcClient that communicates with a node that is online.
+    fn retrieve_ingest_summary_node_mappings(
+        &self,
+    ) -> Result<Vec<IngestSummaryNodeMapping>, OverseerError> {
+        let mut ingest_summary_node_mappings: Vec<IngestSummaryNodeMapping> = Vec::new();
+        for (ingest_client_index, ingest_client) in self.ingest_clients.iter().enumerate() {
+            match ingest_client.get_status() {
+                Ok(ingest_summary) => {
+                    log::trace!(
+                        self.logger,
+                        "Ingest summary retrieved: {:?}",
+                        ingest_summary
+                    );
+                    ingest_summary_node_mappings.push(IngestSummaryNodeMapping {
+                        node_index: ingest_client_index,
+                        ingest_summary,
+                    });
+                }
+                Err(_) => {
+                    let error_message = format!(
+                        "Unable to retrieve ingest summary for node: {}",
+                        ingest_client.get_uri()
+                    );
```

Comment:
it would be better to capture the actual error and format it into this string, so like:

```
                Err(err) => {
                    let error_message = format!(
                        "Unable to retrieve ingest summary for node ({}): {}",
                        ingest_client.get_uri(),
                        err
                    );
```

or something like this

Reply:
Done.